src/main/java/gift/controller/ProductController.java

```diff
@@ -1,32 +1,60 @@
 package gift.controller;
 
+import gift.domain.Option;
 import gift.domain.Product;
+import gift.dto.request.OptionRequest;
 import gift.dto.request.ProductRequest;
+import gift.dto.response.OptionResponse;
 import gift.dto.response.ProductResponse;
 import gift.service.CategoryService;
+import gift.service.OptionService;
 import gift.service.ProductService;
 import jakarta.validation.Valid;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
 import org.springframework.web.bind.annotation.*;
 
+import java.util.List;
+
 @Controller
 @RequestMapping("/api/products")
 public class ProductController {
 
     private final ProductService productService;
     private final CategoryService categoryService;
+    private final OptionService optionService;
 
     @Autowired
-    public ProductController(ProductService productService, CategoryService categoryService) {
+    public ProductController(ProductService productService, CategoryService categoryService, OptionService optionService) {
         this.productService = productService;
         this.categoryService = categoryService;
+        this.optionService = optionService;
+    }
+
+    @GetMapping("/{productId}/options")
+    public ResponseEntity<List<OptionResponse>> getOptions(@PathVariable Long productId) {
+        List<OptionResponse> options = optionService.getOptionsByProductId(productId);
+        return ResponseEntity.ok(options);
+    }
+
+    @PostMapping("/{productId}/options")
+    public ResponseEntity<Option> addOption(@PathVariable Long productId, @ModelAttribute @Valid OptionRequest optionRequest) {
+        Option createdOption = optionService.addOptionToProduct(productId, optionRequest);
+        return ResponseEntity.status(HttpStatus.CREATED).body(createdOption);
     }
 
-    @GetMapping()
+    @DeleteMapping("/{productId}/{optionId}")
```

Comment:
```suggestion
    @DeleteMapping("/{productId}/options/{optionId}")
```

변경하고자 하는 리소스를 적어주는 것도 좋을 것 같아요.

실제로 클라이언트가 요청시에는 host/api/products/1/2 이렇게 요청할텐데, url 이 명확하지 않은 것 같아요.

Reply:
넵 감사합니다!!